platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/hash/MerkleHashBuilder.java

```diff
@@ -142,77 +113,104 @@ public Future<Hash> digestTreeAsync(MerkleNode root) {
         } else if (root.getHash() != null) {
             return new StandardFuture<>(root.getHash());
         } else {
-            final FutureMerkleHash result = new FutureMerkleHash();
-            AtomicInteger activeThreadCount = new AtomicInteger(cpuThreadCount);
-            for (int threadIndex = 0; threadIndex < cpuThreadCount; threadIndex++) {
-                threadPool.execute(createHashingRunnable(threadIndex, activeThreadCount, result, root));
-            }
-            return result;
+            resultFuture = new FutureMerkleHash();
+            ResultTask resultTask = new ResultTask(root, resultFuture);
+            new TraverseTask(root, resultTask).send();
+            return resultFuture;
         }
     }
 
-    /**
-     * Create a thread that will attempt to hash the tree starting at the root.
-     *
-     * @param threadId
-     * 		Used to generate a randomized iteration order
-     */
-    private Runnable createHashingRunnable(
-            final int threadId, AtomicInteger activeThreadCount, final FutureMerkleHash result, final MerkleNode root) {
-
-        return () -> {
-            final MerkleIterator<MerkleNode> it = root.treeIterator()
-                    .setFilter(MerkleHashBuilder::filter)
-                    .setDescendantFilter(MerkleHashBuilder::descendantFilter);
-            if (threadId > 0) {
-                // One thread can iterate in-order, all others will use random order.
-                it.setOrder(POST_ORDERED_DEPTH_FIRST_RANDOM);
```

Comment:
POST_ORDERED_DEPTH_FIRST_RANDOM doesn't seem to be used anywhere after this change, can be removed altogether

Reply:
It's still a meaningful mode in `MerkleIterationOrder` that potentially could be used for other purposes.
Let's leave it until the whole Merkle tree is revisited in the MegaMap context.